src/core/MIGRATION.md

```diff
@@ -1320,6 +1320,13 @@ export class Plugin implements Plugin<PluginSetup, PluginStart> {
   }
 ```
 
+If you want to disable your plugin by default, you could declare the `enabled` flag in plugin config. This is a special Kibana platform key. The platform reads its value and won't create a plugin instance if `enabled: false`. All plugins are considered enabled by default.
```

Comment:
NIT: I would move the last sentence at first position (but then this creates a visible 'default' repetition and would probably need some synonym or rewording)

> All plugins are considered enabled by default. If you want to disable your plugin by default, you could declare the `enabled` flag in plugin config. This is a special Kibana platform key. The platform reads its value and won't create a plugin instance if `enabled: false`.